### ISSUE TITLE
Replace Licence Wordcloud with Sunburst

### ISSUE THREAD

**Issue by aidan-epperly** [LLNL/llnl.github.io#359]
The current word cloud for the licenses does not give a clear picture of the true proportions for repos with a given license. Replacing it with a [sunburst](https://observablehq.com/@d3/sunburst) or [zoomable sunburst](https://observablehq.com/@d3/zoomable-sunburst) would give a better visualization of the true percentages as well as increase the information presentable in the graphic.

**Comment by IanLee1521** [comment]
Agreed! 👍

**Comment by aidan-epperly** [comment]
Label-less sunburst for licenses. Partition based on number of stargazers.
Purple: BSD-3
Pink: GPL-2.0
Salmon: MIT
Orange: BSD-2

**Comment by IanLee1521** [comment]
by "partition based on number of stargazers" do you mean the second level of that graph?

@hauten -- Is this something where should try to adopt some sort of lab color scheme? This is maybe a TID question?

**Comment by aidan-epperly** [comment]
> by "partition based on number of stargazers" do you mean the second level of that graph?

It actually partitions both levels by stargazers. The second layer is partitioned directly and the first layer is partitioned by the sum of its children's stargazers.

> Is this something where should try to adopt some sort of lab color scheme? This is maybe a TID question?

Changing the colors shouldn't be an issue. If you give me a continuous range of colors, that would be best but I can work with a sufficiently large number of discrete colors too.

**Comment by hauten** [comment]
> @hauten -- Is this something where should try to adopt some sort of lab color scheme? This is maybe a TID question?

Apart from Lab Blue and a gray accent, there is no color scheme. ENG (https://engineering.llnl.gov/) and WCI (https://wci.llnl.gov/) have accent colors in the "cool" spectrum. We are redesigning the COMP site completely and will align with Lab Blue. I'll put the One Lab template (managed by TID) link in Slack for reference.

**Comment by aidan-epperly** [comment]
It would be great to get all of the blue on the page in line, there is a bit of an issue with different blues used throughout the page.

**Comment by aidan-epperly** [comment]
Working on a better way to deal with the text bleed. Other than that, thoughts?
![image](https://user-images.githubusercontent.com/29114346/86298298-49813a00-bbb2-11ea-9f32-a4475a3b2b55.png)
![image](https://user-images.githubusercontent.com/29114346/86298331-5a31b000-bbb2-11ea-9142-310e3984604d.png)
![image](https://user-images.githubusercontent.com/29114346/86298362-6d448000-bbb2-11ea-8d15-0d692997f050.png)

**Comment by LRWeber** [comment]
We may want to be careful about how we handle text overlaying on colors; Poor contrast between the background color and foreground text makes it hard to read.

The dark purple specifically stands out as a problem.
(see accessibility analysis https://material.io/resources/color/#!/?view.left=1&view.right=1&primary.color=9C27B0)

**Comment by aidan-epperly** [comment]
I hadn't handled colors yet because of the questions regarding the lab color scheme. I will probably change the text to white and use darker colors.

**Comment by aidan-epperly** [comment]
Now that I look at the tool, I will actually probably do the opposite. From what I can see, black is more visible on average then white is that your understanding @LRWeber ? If that is the case I can use lighter colors. Do you have any idea on where I could get a continuous range of sufficiently light colors?

**Comment by LRWeber** [comment]
I would agree that black text is more ideal, particularly since that's what most of our site is using.
If you are still using D3 (I assume so based on the links in the initial post), https://github.com/d3/d3-scale-chromatic#schemeSet3 looks like a good choice for black text.
Unfortunately, it wouldn't provide an infinite number of colors, but I believe they would at least be sufficient for the licenses.

**Comment by aidan-epperly** [comment]
Alright, thanks. This is the graph with schemeSet3:
![image](https://user-images.githubusercontent.com/29114346/86301533-f790e200-bbba-11ea-929c-550d0de56671.png)

**Comment by LRWeber** [comment]
MUCH more legible! 👍
Even if we decide to tweak colors for theming purposes later, that should make a good placeholder.
(Of course we may find we need a different option to get more discrete colors for other graphs....)

**Comment by IanLee1521** [comment]
I like it!

Is this all repos? Or only the LLNL org? -- I ask because I don't see Spack / ZFS which should be big chunks if we're sizing by stars.

**Comment by IanLee1521** [comment]
which in turn makes me wonder if we **want** to size by stars

**Comment by aidan-epperly** [comment]
> Is this all repos? Or only the LLNL org? -- I ask because I don't see Spack / ZFS which should be big chunks if we're sizing by stars.
Spack and ZFS are in the page, but they use non-standard licenses. 
> which in turn makes me wonder if we want to size by stars
I am going to work on adding a way to change the weighting for the graph. One thing is that if we don't weight the repos somehow, all of the repos will be the same size and I don't think any will be labeled because the slices will be so small.

**Comment by aidan-epperly** [comment]
Fixed word overlap:
![image](https://user-images.githubusercontent.com/29114346/86384500-ad077800-bc43-11ea-8fa1-4cbcdff21a65.png)
![image](https://user-images.githubusercontent.com/29114346/86384548-bdb7ee00-bc43-11ea-8222-dfbdcc928cd2.png)

**Comment by aidan-epperly** [comment]
Weighting by forks:
![image](https://user-images.githubusercontent.com/29114346/86386076-21dbb180-bc46-11ea-9827-2279dc664bc6.png)

**Comment by aidan-epperly** [comment]
Slider changes data:
![image](https://user-images.githubusercontent.com/29114346/86396026-12fcfb00-bc56-11ea-9f84-d742a492790d.png)
![image](https://user-images.githubusercontent.com/29114346/86396075-25773480-bc56-11ea-923b-dd5e6e783d50.png)
![image](https://user-images.githubusercontent.com/29114346/86396097-2c05ac00-bc56-11ea-8600-8ccda0d8ee33.png)
![image](https://user-images.githubusercontent.com/29114346/86396105-332cba00-bc56-11ea-9c56-dbff5717f573.png)

**Comment by aidan-epperly** [comment]
Names that are too long get condensed. Title still full name.
![image](https://user-images.githubusercontent.com/29114346/86396516-e5fd1800-bc56-11ea-9420-fd2232e7410a.png)

**Comment by IanLee1521** [comment]
When you drill down, can it include the License name in the middle? E.g. this last one that's salmon / red / etc.

**Comment by IanLee1521** [comment]
I think this all looks good, so perhaps it's time to get a pull request going?

**Comment by IanLee1521** [comment]
We can always further refine with more pull requests.

**Comment by aidan-epperly** [comment]
> When you drill down, can it include the License name in the middle? E.g. this last one that's salmon / red / etc.

Yes, I can add that and then put in a pr.

**Comment by aidan-epperly** [comment]
@IanLee1521 I changed the slider to have better names and added an 'other' category to the wheel. However, the addition of other meant the addition of even more lag. Here is how it looks now. I am going to do some cleanup in the hopes of significantly reducing the lag, but without that, I don't think this can go live, aside from the slider changes:
![image](https://user-images.githubusercontent.com/29114346/86948412-f18e8a00-c101-11ea-8d06-1e102769c099.png)

**Comment by IanLee1521** [comment]
Yeah, that seems excessive... do we need a toggle for turning off the "other" catagory? Is that a way to solve this?

Another possibility... should we default to the # of repos instead of # of stars?

**Comment by aidan-epperly** [comment]
We are currently using d3 release 4.2, but d3 is on 5.7. d3 5 adds a tool that I would really need if we want to dynamically change the actual data that is shown. I talked to @LRWeber about it getting the newest version on Slack. Once I have that, then I can use d3.join to actually remove data instead of just hiding the paths. Also, as per #371 I fixed the lag problem caused by the addition of "Other" and it is now less laggy than it was to begin with.

I also think that stars is the best way to go, because number of repos is really a crude metric of importance. I am not sure that it matters that a large number of repos are under MIT if the repos aren't actually that popular. The MIT specifically is a great license for smaller projects as far as I have heard from my discussions with my dad on the subject. I think that what really matters are the licenses that people are the most likely to encounter when using LLNL software, and stars is the best metric we have available for that. Also, there are a lot of repos that either aren't kept up well or that no one really uses that would be given equal footing on the repo graph. For context, here is the repo graph:
![image](https://user-images.githubusercontent.com/29114346/86999238-fa657700-c166-11ea-8d52-4126adc05554.png)
